src/lib/schedule.c

```diff
@@ -367,6 +372,13 @@ void schedule_task_complete(struct task *task)
 
 	/* tell any waiter that task has completed */
 	wait_completed(&task->complete);
+
+	/* schedule the same pending task */
+	if (task->pending_cnt) {
+		ticks_per_ms = clock_ms_to_ticks(sch->clock, 1);
```

Comment:
The pending_cnt may be wrong now.
Let's image if pending_cnt is 2, and the state is PENDING. schedule_task() will be called. In _schedule_task(), the pending_cnt will be added again. But in this case, the cnt should not be added, right?